dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/OperationsParamsValidator.java

```diff
@@ -0,0 +1,214 @@
+      Set<OrganisationUnit> orgUnits,
+      OrganisationUnitSelectionMode orgUnitMode,
+      Program program,
+      Function<String, List<OrganisationUnit>> orgUnitDescendants,
+      TrackerAccessManager trackerAccessManager)
+      throws ForbiddenException {
+
+    Set<OrganisationUnit> accessibleOrgUnits = new HashSet<>();
+
+    for (OrganisationUnit orgUnit : orgUnits) {
+      Set<OrganisationUnit> accessibleOrgUnitsFound =
+          switch (orgUnitMode) {
+            case DESCENDANTS -> getAccessibleDescendants(
+                user, program, orgUnitDescendants.apply(orgUnit.getUid()));
+            case CHILDREN -> getAccessibleDescendants(
+                user,
+                program,
+                Stream.concat(Stream.of(orgUnit), orgUnit.getChildren().stream()).toList());
+            case SELECTED -> getSelectedOrgUnits(user, program, orgUnit, trackerAccessManager);
+            default -> Collections.emptySet();
+          };
+
+      if (accessibleOrgUnitsFound.isEmpty()) {
+        throw new ForbiddenException("User does not have access to orgUnit: " + orgUnit.getUid());
+      }
+
+      accessibleOrgUnits.addAll(accessibleOrgUnitsFound);
+    }
+
+    if (orgUnitMode == CAPTURE) {
+      return new HashSet<>(user.getOrganisationUnits());
+    } else if (orgUnitMode == ACCESSIBLE) {
+      return getAccessibleOrgUnits(user, program);
+    }
+
+    return accessibleOrgUnits;
+  }
+
+  private static Set<OrganisationUnit> getSelectedOrgUnits(
+      User user,
+      Program program,
+      OrganisationUnit orgUnit,
+      TrackerAccessManager trackerAccessManager) {
+    return trackerAccessManager.canAccess(user, program, orgUnit)
+        ? Set.of(orgUnit)
+        : Collections.emptySet();
+  }
+
+  private static Set<OrganisationUnit> getAccessibleOrgUnits(User user, Program program) {
+    return isProgramAccessRestricted(program)
+        ? new HashSet<>(user.getOrganisationUnits())
+        : new HashSet<>(user.getTeiSearchOrganisationUnitsWithFallback());
+  }
+
+  /**
+   * Returns the org units whose path is contained in the user search or capture scope org unit. If
+   * there's a match, it means the user org unit is at the same level or above the supplied org
+   * unit.
+   *
+   * @param user the user to check the access of
+   * @param program the program the user wants to access to
+   * @param orgUnits the org units to check if the user has access to
+   * @return a list with the org units the user has access to
+   */
+  private static Set<OrganisationUnit> getAccessibleDescendants(
+      User user, Program program, List<OrganisationUnit> orgUnits) {
+    if (orgUnits.isEmpty()) {
+      return Collections.emptySet();
+    }
+
+    Set<OrganisationUnit> userOrgUnits =
+        isProgramAccessRestricted(program)
+            ? user.getOrganisationUnits()
+            : user.getTeiSearchOrganisationUnits();
+
+    return orgUnits.stream()
+        .filter(
+            availableOrgUnit ->
```

Comment:
nit: try to use shorter var names in lambdas like `o` for an orgUnit. Personally I find lambdas harder to read with var names I would use in a non-lambda form.

Reply:
I would normally agree, but in this case I think writing a descriptive name helps understand what this lambda does. 
It iterates two different list of org units, and in my opinion, having a short var name would make it difficult to understand the list we are dealing with each time.

Reply:
The name of the collection var should give away what they are IMHO. Not sure if I remember correctly but maybe it was in the other PR that you had similar code with two different lambdas that were actually doing exactly the same but used different names for the lambda var. That itself obscures the lambda and prevents us from extracting generic functions 😅